tests/e2e/utils/KubernetesCommandLineToolsExecutor.ts

```diff
@@ -27,11 +28,15 @@ export class KubernetesCommandLineToolsExecutor extends ShellExecutor {
     // login to Openshift cluster with username and password
     loginToOcp(): void {
         if (this.KUBERNETES_COMMAND_LINE_TOOL === KubernetesCommandLineTool.OC) {
-            Logger.debug(`${this.getLoggingName(this.loginToOcp.name)}: Login to the "OC" client`);
+            Logger.debug(`${this.getLoggingName(this.loginToOcp.name)}: Login to the "OC" client.`);
             const url: string = this.getServerUrl();
-            Logger.debug(url, TestConstants.TS_SELENIUM_OCP_USERNAME);
-            exec(`sleep 5
-            oc login --server=${url} -u=${TestConstants.TS_SELENIUM_OCP_USERNAME} -p=${TestConstants.TS_SELENIUM_OCP_PASSWORD} --insecure-skip-tls-verify`);
+            const isUserLoggedIn: ShellString = this.execWithLog('oc whoami && oc whoami --show-server=true');
```

Comment:
'oc whoami --show-server=true' would be enough, IMHO.

Also name `isUserLoggedIn` doesn't comply with value.
**Expected value**: true or false
**Actual value**: string with output of `oc whoami --show-server=true` like `admin`, or error message if user hasn't logged into the cluster.

I would suggest to extract the code into the `isUserLoggedIn()` method:
private isUserLoggedIn(): boolean {
    const whoamiCommandOutput: ShellString = this.execWithLog('oc whoami --show-server=true');

     return whoamiCommandOutput.stdout.includes(TestConstants.TS_SELENIUM_OCP_USERNAME) && whoamiCommandOutput.stdout.includes(url)
}

Reply:
Agree

Reply:
Thanks for the fix.